File: app/src/main/java/forpdateam/ru/forpda/api/search/models/SearchItem.java
package forpdateam.ru.forpda.api.search.models;

import forpdateam.ru.forpda.api.BaseForumPost;
import forpdateam.ru.forpda.api.search.interfaces.ISearchItem;

/**
 * Created by radiationx on 01.02.17.
 */

public class SearchItem extends BaseForumPost implements ISearchItem {
    private String title, desc, imageUrl;

    @Override
    public String getImageUrl() {
        return imageUrl;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public String getDesc() {
        return desc;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/entity/remote/search/SearchItem.kt
package forpdateam.ru.forpda.entity.remote.search

import forpdateam.ru.forpda.entity.remote.ForumPost
import forpdateam.ru.forpda.entity.remote.others.user.User
import ru.radiationx.coretypes.ArticleId
import ru.radiationx.coretypes.ForumId
import ru.radiationx.coretypes.TopicId

/**
 * Created by radiationx on 01.02.17.
 */

sealed interface SearchItem {

    data class News(
        val id: ArticleId,
        val imageUrl: String,
        val date: String,
        val user: User,
        val title: String,
        val body: String,
    ) : SearchItem

    data class Topic(
        val topicId: TopicId,
        val title: String,
        val desc: String,
        val forumId: ForumId,
        val user: User,
        val date: String,
    ) : SearchItem

    data class Post(
        val title: String,
        val post: ForumPost
    ) : SearchItem
}
